apps/finance/contracts/Finance.sol

```diff
@@ -139,60 +134,43 @@ contract Finance is AragonApp, ERC677Receiver {
     * @param _amount Amount of tokens sent
     * @param _reference Reason for payment
     */
-    function deposit(ERC20 _token, uint256 _amount, string _reference) external transitionsPeriod {
+    function deposit(address _token, uint256 _amount, string _reference) external transitionsPeriod {
         _recordIncomingTransaction(
             _token,
             msg.sender,
             _amount,
             _reference
         );
-        require(_token.transferFrom(msg.sender, address(vault), _amount));
+        // first we need to get the tokens to Finance
+        ERC20(_token).transferFrom(msg.sender, this, _amount);
+        // and then approve them to vault
+        ERC20(_token).approve(address(vault), _amount);
+        // finally we can deposit them
+        vault.deposit(_token, this, _amount, new bytes(0));
+        //vault.deposit(_token, msg.sender, _amount, new bytes(0));
```

Comment:
Should this commented out line be removed?

Reply:
sure, thanks!